playbooks/destroy_infrastructure.yml

```diff
@@ -52,8 +52,19 @@
   connection: local
   gather_facts: false
   tasks:
-    - name: Delete the resource group and all resources in it
-      azure.azcollection.azure_rm_resourcegroup:
+    - name: Get resources in the resource group
+      azure.azcollection.azure_rm_resourcegroup_info:
         name: "{{ resource_group }}"
+        list_resources: true
+      register: resource_group_info
+
+    - name: Filter resources by deployment_id tag
+      ansible.builtin.set_fact:
+        target_resources: "{{ resource_group_info.resourcegroups[0].resources | selectattr('tags.deployment', 'defined') |
+          selectattr('tags.deployment', 'eq', 'aap-infrastructure-' + deployment_id) | map(attribute='id') | list }}"
+
+    - name: Delete resources
+      azure.azcollection.azure_rm_resource:
+        url: "{{ item }}"
```

Comment:
```suggestion
        url: "https://management.azure.com{{ item }}"
```
Then you might need to move this play into it's own task file and do something like this so it iterates over the same tasks until things are cleaned up.
https://github.com/ansible/ansible/issues/46203#issuecomment-556013701

This is because there will be conflicts with order of deletions. You might need to add a small wait in the rescue tasks as well.

Reply:
added suggested change